eclair-core/src/main/scala/fr/acinq/eclair/io/PeerConnection.scala

```diff
@@ -456,7 +467,7 @@ object PeerConnection {
   sealed trait Data
   sealed trait HasTransport extends Data { def transport: ActorRef }
   case object Nothing extends Data
-  case class AuthenticatingData(pendingAuth: PendingAuth, transport: ActorRef) extends Data
+  case class AuthenticatingData(pendingAuth: PendingAuth, transport: ActorRef) extends Data with HasTransport
```

Comment:
nit: `HasTransport` already extends `Data`, so you don't need `extends Data with HasTransport` (and same below).

Reply:
I know, that's because I have used the cake pattern too much.

Reply:
Done with df54c287d

Reply:
What does that cake pattern bring here (apart from being less explicit than `sealed trait HasTransport extends Data`)?

Reply:
Not much: it conveys the idea that `HasTransport` is just a behavior and not a definition, which is closer to how I see it.

Reply:
Got it, I'm not very familiar with that pattern

Reply:
In other words you have to use `Data with HasTransport`, you can't use `HasTransport` by itself.